src/Core/src/Stream/ResponseBodyStream.php

```diff
@@ -33,45 +44,47 @@ public function __toString()
      */
     public function getChunks(): iterable
     {
+        if (null !== $this->fallback) {
+            return $this->fallback->getChunks();
+        }
+
+        if (null === $this->resource) {
+            $this->resource = \fopen('php://temp', 'rb+');
+        }
         foreach ($this->responseStream as $chunk) {
-            yield $chunk->getContent();
+            $chunkContent = $chunk->getContent();
+            \fwrite($this->resource, $chunkContent);
+            yield $chunkContent;
         }
+
+        $this->fallback = new ResponseBodyResourceStream($this->resource);
```

Comment:
What if I start reading the stream. Stop half ways and restart?

Reply:
indeed... we have to consume the entire `responseStream`.. That defect the purpose of this class. we don't stream anythings. This should be deprecated an replaced by `ResponseBodyResourceStream`

Reply:
Im happy to remove `ResponseBodyStream` if we always can use `ResponseBodyResourceStream` . 

If we cant, we should add a note that `ResponseBodyStream` is only a BC layer and document the edge case if you stop half ways.

Reply:
but we don't have choice... Moreover this class should not be used, people using a standard httpClient will use the other one.

Reply:
That is okey. But it should be documented on the class. 
I was trying to say that by:

> we should add a note that ResponseBodyStream is only a BC layer and document the edge case if you stop half ways.

Reply:
comment added